superset/utils/dashboard_import_export.py

```diff
@@ -38,12 +40,23 @@ def import_dashboards(session, data_stream, import_time=None):
     session.commit()
 
 
-def export_dashboards(session):
+def export_dashboards(session, dashboard_ids=None, dashboard_titles=None):
     """Returns all dashboards metadata as a json dump"""
     logging.info('Starting export')
-    dashboards = session.query(Dashboard)
-    dashboard_ids = []
-    for dashboard in dashboards:
-        dashboard_ids.append(dashboard.id)
-    data = Dashboard.export_dashboards(dashboard_ids)
+    export_dashboard_ids = []
+
+    session = db.session()
+    query = session.query(Dashboard)
+    if dashboard_ids or dashboard_titles:
+        query = query.filter(Dashboard.id.in_(dashboard_ids) |
+                             Dashboard.dashboard_title.in_(dashboard_titles))
+
+    export_dashboard_ids = [d.id for d in query.all()]
+
+    data = {}
+    if not export_dashboard_ids:
+        logging.error('No dashboards found!')
+        raise DashboardNotFoundException('No dashboards found!')
+    else:
+        data = Dashboard.export_dashboards(export_dashboard_ids) if export_dashboard_ids else {}
```

Comment:
No need for the ternary `if` here anymore I think

Reply:
k